pkg/apis/azure/storage/v1alpha1/account_test.go

```diff
@@ -0,0 +1,1048 @@
+/*
+Copyright 2018 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"testing"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/storage/mgmt/2017-06-01/storage"
+	"github.com/Azure/go-autorest/autorest/date"
+	"github.com/Azure/go-autorest/autorest/to"
+	"github.com/go-test/deep"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func Test_newCustomDomain(t *testing.T) {
```

Comment:
great coverage and cases for all these types and constructors.  Did you do these all these tests/cases by hand?

Reply:
Kind'a, my IDE (Goland) auto-generates unit test scaffolding, but it is still requires some updates.

Reply:
Oooh, I might want you to show me this some time :)

Reply:
https://github.com/cweill/gotests

My guess is that GoLand uses this under the hood, as does VSCode's Go plugin.

(Also, TIL Gogland is called GoLand now.)